dashboard/app/models/user_level.rb

```diff
@@ -129,13 +129,16 @@ def self.update_lockable_state(user_id, level_id, script_id, locked, readonly_an
     # no need to create a level if it's just going to be locked
     return if !user_level.persisted? && locked
 
-    user_level.update!(
+    user_level.assign_attributes(
       submitted: locked || readonly_answers,
       readonly_answers: !locked && readonly_answers,
       unlocked_at: locked ? nil : Time.now,
       # level_group, which is the only levels that we lock, always sets best_result to 100 when complete
       best_result: (locked || readonly_answers) ? ActivityConstants::BEST_PASS_RESULT : user_level.best_result
     )
+
+    # preserve updated_at, which represents the user's submission timestamp.
+    user_level.save!(touch: false)
```

Comment:
@aoby , could I please have a sanity check from a rails perspective that `assign_attributes` + `save!` is equivalent to `update!`? That step is intended to be a pure refactor (no behavior change), and the `touch: false` param is supposed to be the part that implements the behavior change.